### ISSUE TITLE
Add melodic (aka "jazz") minor scale

### ISSUE THREAD

**Issue by spittinglama** [kilpatrickaudio/CARBON#91]
Many of the included scales are simply modes of the major scale.  I'd like to see a few more unique scales. But the absence of the Melodic Minor scale seems a significant omission. Though I find the included modes of the major scale handy I think root scales should take precedence over modes for inclusion in applications such as this.  Melodic minor is identical to the Dorian scale except for having a Maj 7th instead of a Min 7th. R Maj 2 min 3 P4 P5 Maj 6 Maj 7th.   C D Eb F G A B     Following that, the inclusion of a could hexatonic scales would be super swell - particularly Dorian and Major hex.

**Comment by kilpatrickaudio** [comment]
Yes there definitely could always be more scales. The reason that the melodic minor wasn't included is because normally it is different in the ascending and descending direction, but the scale processor in CARBON doesn't know how to tell which form to use. We can add the ascending version (which is the unique part) and will consider this in the next version.
